File: Loenn/entities/force_jump_crystal.lua
local utils = require "utils"
local drawableSprite = require "structs.drawable_sprite"

local forceJumpCrystal = {}

forceJumpCrystal.name = "vitellary/forcejumpcrystal"
forceJumpCrystal.depth = -100

forceJumpCrystal.fieldInformation = {
    respawnTime = {
        minimumValue = 0.0
    }
}

forceJumpCrystal.placements = {
    {
        name = "force_jump_crystal",
        data = {
            oneUse = false,
            respawnTime = 2.5,
            holdJump = true
        }
    }
}

function forceJumpCrystal.sprite(room, entity)
    local left = drawableSprite.fromTexture("objects/crystals/forcejump/idle00", entity)
    left:addPosition(-4, 0)

    local right = drawableSprite.fromTexture("objects/crystals/forcejump/idle00", entity)
    right:addPosition(4, 0)
    right:setScale(-1, 1)

    return {left, right}
end

function forceJumpCrystal.selection(room, entity)
    local x, y = entity.x or 0, entity.y or 0
    return utils.rectangle(x - 8, y - 8, 16, 16)
end

return forceJumpCrystal

File: Ahorn/entities/forceJumpCrystal.jl
module ForceJumpCrystal

using ..Ahorn, Maple

@mapdef Entity "vitellary/forcejumpcrystal" Crystal(x::Integer, y::Integer, oneUse::Bool=false,
    respawnTime::Number=2.5, holdJump::Bool=true)

const placements = Ahorn.PlacementDict(
    "Force Jump Crystal (Crystalline)" => Ahorn.EntityPlacement(
        Crystal
    )
)

function Ahorn.selection(entity::Crystal)
    x, y = Ahorn.position(entity)

    return Ahorn.Rectangle(x-8, y-8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Crystal)
    Ahorn.drawSprite(ctx, "objects/crystals/forcejump/idle00.png", -4, 0, jx=0.5, jy=0.5)
    Ahorn.drawSprite(ctx, "objects/crystals/forcejump/idle02.png", 4, 0, jx=0.5, jy=0.5, sx=-1)
end

end